base/v0_5_exp/translate_test.go

```diff
@@ -1699,6 +1699,293 @@ func TestTranslateKernelArguments(t *testing.T) {
+			sshKeyDir,
+			filepath.Join(sshKeyDir, "id_ed25519.pub"),
+		},
+		{
+			"empty ssh_keys_local file",
+			PasswdUser{SSHAuthorizedKeysLocal: []string{sshKeyEmptyFileName}},
+			types.PasswdUser{},
+			"error at $.ssh_authorized_keys_local: local file to be embedded must not be empty\n",
+			sshKeyDir,
+			"",
+		},
+		{
+			"blank ssh_keys_local file",
+			PasswdUser{SSHAuthorizedKeysLocal: []string{sshKeyBlankFileName}},
+			types.PasswdUser{},
+			"error at $.ssh_authorized_keys_local: local file to be embedded must not be empty\n",
+			sshKeyDir,
+			"",
+		},
+		{
+			"missing embed directory",
+			PasswdUser{SSHAuthorizedKeysLocal: []string{sshKeyFileName}},
+			types.PasswdUser{},
+			"error at $.ssh_authorized_keys_local: local file paths are relative to a files directory that must be specified with -d/--files-dir\n",
+			"",
+			"",
+		},
+		{
+			"wrong embed directory",
+			PasswdUser{SSHAuthorizedKeysLocal: []string{sshKeyFileName}},
+			types.PasswdUser{},
+			"error at $.ssh_authorized_keys_local: open %FilePath%:",
+			randomDir,
+			filepath.Join(randomDir, "id_rsa.pub"),
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			actual, translations, r := translatePasswdUser(test.in, common.TranslateOptions{FilesDir: test.fileDir})
+			assert.Equal(t, test.out, actual, "translation mismatch")
+			expectedReport := strings.ReplaceAll(test.report, "%FilePath%", test.filePath)
+			assert.True(t, strings.HasPrefix(r.String(), expectedReport), "bad report")
+			assert.NoError(t, translations.DebugVerifyCoverage(actual), "incomplete TranslationSet coverage")
+		})
+	}
+}
+
+// TestTranslateSystemdLocalUnits tests translating the butane systemd.units[i].contents_local entries to ignition systemd.units[i].contents entries.
+func TestTranslateSystemdLocalUnits(t *testing.T) {
+	unitDir := t.TempDir()
+	randomDir := t.TempDir()
+	var unitName = "dummy.service"
+	var unitDefinition = "[Service]\nExecStart=/bin/false\n"
+	var unitEmptyFileName = "empty.service"
+	var unitEmptyDefinition = ""
+	var unitNonExistingFileName = "random.service"
+
+	err := ioutil.WriteFile(filepath.Join(unitDir, unitName), []byte(unitDefinition), 0644)
+	if err != nil {
+		t.Error(err)
+	}
+	err = ioutil.WriteFile(filepath.Join(unitDir, unitEmptyFileName), []byte(unitEmptyDefinition), 0644)
+	if err != nil {
+		t.Error(err)
+	}
+
+	tests := []struct {
+		name     string
+		in       Unit
+		out      types.Unit
+		report   string
+		fileDir  string
+		filePath string
+	}{
+		{
+			"empty unit",
+			Unit{},
+			types.Unit{},
+			"",
+			"",
+			"",
+		},
+		{
+			"valid contents",
+			Unit{Contents: &unitDefinition, Name: unitName},
+			types.Unit{Contents: &unitDefinition, Name: unitName},
+			"",
+			"",
+			"",
+		},
+		{
+			"valid contents_local",
+			Unit{ContentsLocal: &unitName, Name: unitName},
+			types.Unit{Contents: &unitDefinition, Name: unitName},
+			"",
+			unitDir,
+			filepath.Join(unitDir, unitName),
+		},
```

Comment:
Need a case that has both `Contents` and `ContentsLocal`, for each of units and dropins.